lib/mjsonwp/mjsonwp.js

```diff
@@ -245,6 +245,11 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
         driverRes = null;
       }
 
+      // if the status is not 0,  throw the appropriate error for status code.
+      if (util.hasValue(driverRes) && util.hasValue(driverRes.status) && parseInt(driverRes.status, 10) !== 0) {
+        throw errorFromCode(driverRes.status, driverRes.value);
```

Comment:
The error handling code is in the exception handler [below](https://github.com/appium/appium-base-driver/blob/master/lib/mjsonwp/mjsonwp.js#L258-L268). Some drivers (e.g., WDA, and, at the moment, the UiAutomator2 driver) don't respond with exceptions, but set the status codes, so we would end up with a `200` http status with a Selenium error inside, which confuses the clients.